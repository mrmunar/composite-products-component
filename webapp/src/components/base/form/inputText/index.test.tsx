import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import InputText from './';

afterEach(cleanup);

test('should take a snapshot', () => {
    const { baseElement } = render(<InputText />);
    expect(baseElement).toMatchSnapshot();
});

test('can set a value', () => {
    const { getByTestId } = render(<InputText value="Test Value" />);
    expect(getByTestId('input-text')).toHaveValue('Test Value');
});

test('can change a value', () => {
    const { getByTestId } = render(<InputText onChange={() => {}} value="Test Value" />);
    fireEvent.change(getByTestId('input-text'), { target: { value: '23' } })
    expect(getByTestId('input-text')).toHaveValue('23');
});


