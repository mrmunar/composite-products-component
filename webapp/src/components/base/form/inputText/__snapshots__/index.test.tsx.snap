// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<body>
  <div>
    <input
      class="form-control"
      data-testid="input-text"
      type="text"
      value=""
    />
  </div>
</body>
`;
